spec/active_force/sobject/includes_spec.rb

```diff
@@ -286,5 +286,60 @@ module ActiveForce
         end
       end
     end
+
+    describe '.includes with nested associations' do
```

Comment:
I would like to see quite a few more test cases here.  Some that come to mind (although correct me if I'm misunderstanding and these are already covered):

- multiple nested belongs_to associations
- has_many/has_one nested within a belongs_to
- belongs_to nested within has_many/has_one
- an Array within a Hash (`.includes(a: [:b, :c])`)
- a Hash within a Hash (`.includes(a: { b: :c })`)
- nested invalid association key `.includes(a: :invalid)`

And I think we want the assertions of the query result mapping back into SObjects to be more probative.  For example, we want to check that the attribute values are actually assigned; not just that the SObject types are correct.

Reply:
Added more specs to cover all these scenarios

Reply:
I might have missed it, but was there a multiple nested belongs_to association in there?  Just want to make sure we have that `Rel1.Rel2.Id` case correct, since it's a bit tricky.